pkg/brokerpak/fetch.go

```diff
@@ -25,7 +40,91 @@ func fetchArchive(src, dest string) error {
 		Src:           src,
 		Dst:           dest,
 		Mode:          getter.ClientModeFile,
-		Getters:       getter.Getters,
 		Decompressors: map[string]getter.Decompressor{},
 	}).Get()
 }
+
+// fetchBrokerpak downloads a local or remote brokerpak; brokerpaks can be
+// fetched remotely using the gs:// prefix which will load them from a
+// Cloud Storage bucket with the broker's credentials.
+// Relative paths are resolved relative to the executable.
+func fetchBrokerpak(src, dest string) error {
+	execWd := filepath.Dir(os.Args[0])
+	execDir, err := filepath.Abs(execWd)
+	if err != nil {
+		return fmt.Errorf("couldn't turn dir %q into abs path: %v", execWd, err)
+	}
+
+	getters := map[string]getter.Getter{}
+	for k, g := range getter.Getters {
+		getters[k] = g
+	}
+	getters["gs"] = &gsGetter{}
+
+	return (&getter.Client{
```

Comment:
This method seems to have two responsibilities: 1. initialize the client 2. fetch the broker pak. If you separated these concerns and had the client be passed in as an argument, it would improve orthogonality and allow for unit testing the logic of this module via mocking the client.